### DISCUSSION TITLE
Declare partial derivatives

### DISCUSSION THREAD

**Discussion by Zcaic** [OpenMDAO/OpenMDAO#3160] [Q&A]
### Desired capability or behavior.

Hellow, I have a input `x` with shape (10,), and the output `f` with shape (10,). And thus the shape of  Jacobian is:
```math
\frac{\partial f}{\partial x}=\left[ \begin{matrix}
	j_{00}&		j_{01}&		\cdots&		j_{09}\\
	j_{10}&		j_{11}&		\cdots&		j_{19}\\
	\vdots&		\vdots&		\ddots&		\vdots\\
	j_{90}&		j_{91}&		\cdots&		j_{99}\\
\end{matrix} \right]
```
If I know the values ​​of part of the Jacobian matrix, like:
```math
\left[ \begin{matrix}
	j_{00}&		j_{01}&		\cdots&		j_{08}\\
	j_{10}&		j_{11}&		\cdots&		j_{18}\\
	\vdots&		\vdots&		\ddots&		\vdots\\
	j_{90}&		j_{91}&		\cdots&		j_{98}\\
\end{matrix} \right] =2*I
```
I just want to use `finite difference` to solve:
```math
\left[ \begin{array}{l}
	j_{09}\\
	j_{19}\\
	\vdots\\
	j_{99}\\
\end{array} \right] 
```

I noticed that the parameters `rows` , `cols` and `val`  of `declare_partials` doesn't seem to handle this kind of problem. 
Is there any way to solve my problem?

**Comment by robfalck** [comment]
It sounds like your `x` is functionally two variables packed into one.  If you want to mix the use of analytic and partial derivatives in this case, then my solution would probably be to create two components.

The first computes $f_1$, the first 9 elements of $f$, without the contribution of $x_9$. This component accepts the first 9 elements of $x$ as input.  You can pass only those elements using `src_indices=np.arange(9, dtype=int)` in the call to `connect`.

The second computes $f$, accepting $f_1$ and $x_9$ as inputs. You can FD/CS this component, and OpenMDAO will assemble the derivatives correctly for you.

**Comment by Zcaic** [comment]
Okay, thanks for your answer.
